repo_db_mirror_test.go

```diff
@@ -0,0 +1,301 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+//https://gist.github.com/maximilien/328c9ac19ab0a158a8df  slighlty modified to create a fake package
+//it expects to receive as input where to place the tarball(tarballFilePath)
+// and an array of structs, where the first value is the package name, and the second one is its content
+//Arch dbs are structured as packagename-version-subversion/desc , where desc contains all the relevant info
+type testTarDB struct {
+	PkgName string
+	Content string
+}
+
+//testing content for a db
+func getTestTarDB() []testTarDB {
+	return []testTarDB{
+		testTarDB{PkgName: "acl-2.3.1-1",
+			Content: `%FILENAME%
```

Comment:
I don't know if there is a better way to do this, e.g. by adding a file which can be easily accessible from tests